### PR TITLE
Expose number of open connections via JMX

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: blackbox/docs/admin/monitoring.rst [hunk 1]
```diff
@@ -165,6 +165,35 @@ NodeInfo can be accessed using the JMX MBean object name
...
+| Name                 | Description                                             |
++======================+=========================================================+
+| ``HttpOpen``         | The number of currently established connections via     |
+|                      | HTTP                                                    |
++----------------------+---------------------------------------------------------+
+| ``HttpTotal``        | The number of total connections established via HTTP    |
+|                      | over the life time of a node                            |
++----------------------+---------------------------------------------------------+
+| ``PsqlOpen``         | The number of currently established connections via     |
+|                      | postgres protocol                                       |
```
[the remark this answers is not in the document]
Comment (nslater) [reply blackbox/docs/admin/monitoring.rst]: "the" is correct. please also use "PostgreSQL" in full

File: blackbox/docs/admin/monitoring.rst [hunk 2]
```diff
@@ -165,6 +165,35 @@ NodeInfo can be accessed using the JMX MBean object name
 
    Provides the human friendly name of the node
 
+Connections MBean
+-----------------
+
+The ``Connections`` MBean exposes information about the opened connections to a
```
[the remark this answers is not in the document]
Comment (nslater) [reply blackbox/docs/admin/monitoring.rst]: if the nodes are remote, I would change this to:

> exposes information about any open connections to CrateDB nodes.

[the remark this answers is not in the document]
Comment (mfussenegger) [reply blackbox/docs/admin/monitoring.rst]: It's not a sum across nodes but each node provides its own open connections independently.
I think both `of a node` or `to a node` fits.

[the remark this answers is not in the document]
Comment (nslater) [reply blackbox/docs/admin/monitoring.rst]: how about:

> exposes information about any open connections to a CrateDB node.

File: blackbox/docs/admin/monitoring.rst [hunk 3]
```diff
@@ -165,6 +165,35 @@ NodeInfo can be accessed using the JMX MBean object name
...
+| Name                 | Description                                             |
++======================+=========================================================+
+| ``HttpOpen``         | The number of currently established connections via     |
+|                      | HTTP                                                    |
++----------------------+---------------------------------------------------------+
+| ``HttpTotal``        | The number of total connections established via HTTP    |
+|                      | over the life time of a node                            |
++----------------------+---------------------------------------------------------+
+| ``PsqlOpen``         | The number of currently established connections via     |
+|                      | PostgreSQL protocol                                     |
```
Comment (nslater) [inline blackbox/docs/admin/monitoring.rst]: "via HTTP" is correct

but this should say "via the PostgreSQL protocol"

if we were spelling out HTTP, I would also suggest "via the HyperText Transfer Protocol"

I have done some research to try and find a definite rule that explains this but I have not found anything. all I know is that this is what sounds correct and I wish I could explain it

I think it has something to do with the fact that when you talk about the Foo Bar Protocol you are talking about a specific, named, countable, thing (with a specification doc and maybe a wiki page etc etc)

but when you turn it into an initialism, in this context, "FBP" (for instance) becomes a generic uncountable noun phrase. and generic or uncountable noun phrases are not prefixed with the definite article ("the")

compare "the Graphics Interchange Format" with "GIF files" or "encode it as a GIF"

compare the difference between "I like the sunny weather" (which implicitly is understood to refer to the specific sunny weather today) and "I like sunny weather" which refers to "sunny weather" as a generally understood type of thing

I think it really hinges on whether "protocol" (or "format" as per the GIF example) is a part of the sentence or hidden in the abbreviation. when it is hidden, these noun phrases become generic. but when you uses it in the sentence, it forces it into a countable noun

sorry. I wish I could explain better. I hope this helps!

Comment (mfussenegger) [reply blackbox/docs/admin/monitoring.rst]: Thanks for the explanation :+1:

**Comment by mfussenegger** [comment]
@seut pushed a fixup

**Pull request merged**